manifests/worker.pp

```diff
@@ -101,9 +119,23 @@
     }
   }
 
+  # Calculate the amount of memory to make available to Mesos tasks
+  $mesos_requested_mem = floor($::memorysize_mb - $mesos_reserved_mem)
+  if $mesos_requested_mem < $mesos_minimum_mem {
+    $mesos_mem_resources = {
+      'mem(*)' => $mesos_requested_mem
+    }
+  } else {
+    $mesos_mem_resources = {}
+    warning("The requested memory (${mesos_requested_mem}MB) for Mesos is less
+      than the minimum amount allowed (${mesos_minimum_mem}MB). The default
+      Mesos memory resources will be made available instead.")
+  }
```

Comment:
I think this check is the wrong way around. Is it possible to write a test for this?